### PR TITLE
tsuba: subsume parquet parsing code

### PR DESCRIPTION
**Pull request by tylershunt**
Split `PropertyFileGraph` into a graph engine part (called `PropertyFileGraph`) and a storage engine part (called `RDG`). RDG's are simple structs embedded in `PropertyFileGraph` that are populated by tsuba, conversely the graph engine can fill an RDG struct and pass it back to the storage engine to be written out.

Graph names are the same as the name for the meta-data file right now.

The metadata struct is now opaque to the graph engine and controlled by tsuba so that we can do things like enforce capabilities down the line.

Signed-off-by: Tyler Hunt <thunt@katanagraph.com>

### CODE REVIEW

File: libgalois/include/galois/graphs/PropertyFileGraph.h [hunk 1]
```diff
@@ -60,27 +56,27 @@ struct MetadataImpl;
...
 public:
   PropertyFileGraph();
+  PropertyFileGraph(tsuba::RDG&& rdg);
 
-  // Declare destructor out-of-line to accomodate incomplete type MetadataImpl.
+  // Declare destructor out-of-line to accomodate incomplete type
+  // MetadataImpl.
   ~PropertyFileGraph();
 
+  static outcome::std_result<std::shared_ptr<PropertyFileGraph>>
```
Comment (ddn0) [inline libgalois/include/galois/graphs/PropertyFileGraph.h:77]: why not `Result`?

File: libtsuba/src/FileView.cpp [hunk 2]
```diff
@@ -11,18 +11,17 @@
 #include "galois/Logging.h"
 #include "tsuba/tsuba.h"
 
-// Suuuuuuuuuuuuuuuuuuuuuuuuppppppppppppppppeeeeeeeeeeerrrrrrrrrrrr
```
Comment (ddn0) [inline libtsuba/src/FileView.cpp:14]: Easter egg?

Comment (tylershunt) [reply libtsuba/src/FileView.cpp:14]: Git blame said that honor belongs to @scober . I think he was testing to see how closely I read the code and I totally failed that test 👎.

File: libtsuba/src/RDG.cpp [hunk 3]
```diff
@@ -0,0 +1,722 @@
...
+  values.emplace_back(handle.topology_path);
+
+  for (const auto& v : handle.node_properties) {
+    keys.emplace_back(node_property_name_key);
+    values.emplace_back(v.name);
+    keys.emplace_back(node_property_path_key);
+    values.emplace_back(v.path);
+  }
+
+  for (const auto& v : handle.node_properties) {
```
Comment (ddn0) [inline libtsuba/src/RDG.cpp:461]: `handle.edge_properties`

File: libtsuba/src/FileView.cpp (hunk 2 above)
Comment (scober) [reply libtsuba/src/FileView.cpp:14]: I was trying to run clang-format in my editor on saves and just running some tests. But if @tylershunt wants to take responsibility, who am I to stop him?

File: libtsuba/src/RDG.cpp (hunk 3 above)
Comment (tylershunt) [reply libtsuba/src/RDG.cpp]: damn! good eye.

File: libtsuba/src/FileView.cpp (hunk 2 above)
Comment (tylershunt) [reply libtsuba/src/FileView.cpp:14]: 😺

**Pull request merged**